Content.Server/Radio/Components/RadioJammerComponent.cs

```diff
@@ -9,12 +9,46 @@ namespace Content.Server.Radio.Components;
 [Access(typeof(JammerSystem))]
 public sealed partial class RadioJammerComponent : Component
 {
-    [DataField("range"), ViewVariables(VVAccess.ReadWrite)]
-    public float Range = 8f;
+    /// <summary>
+    /// Range of the jammer when on high power
+    /// </summary>
+    [DataField("highPowerRange"), ViewVariables(VVAccess.ReadOnly)]
+    public float HighPowerRange = 12f;
+
+    /// <summary>
+    /// Range of the jammer when on medium power
+    /// </summary>
+    [DataField("mediumPowerRange"), ViewVariables(VVAccess.ReadOnly)]
+    public float MediumPowerRange = 6f;
+
+    /// <summary>
+    /// Range of the jammer when on low power
+    /// </summary>
+    [DataField("lowPowerRange"), ViewVariables(VVAccess.ReadOnly)]
+    public float LowPowerRange = 2.5f;
+
+    /// <summary>
+    /// Power usage per second when on high power
+    /// </summary>
+    [DataField("highPowerWattage"), ViewVariables(VVAccess.ReadOnly)]
+    public float HighPowerWattage = 12f;
+
+    /// <summary>
+    /// Power usage per second when on medium power
+    /// </summary>
+    [DataField("mediumPowerWattage"), ViewVariables(VVAccess.ReadOnly)]
+    public float MediumPowerWattage = 2f;
 
     /// <summary>
-    /// Power usage per second when enabled
+    /// Power usage per second when on low power
     /// </summary>
-    [DataField("wattage"), ViewVariables(VVAccess.ReadWrite)]
-    public float Wattage = 2f;
+    [DataField("lowPowerWattage"), ViewVariables(VVAccess.ReadOnly)]
+    public float LowPowerWattage = 1f;
```

Comment:
Wouldn't it be better to use an enum for this?

Reply:
Yes that makes sense! I'm probably missing something but I'm running into an issue:
Making the selectedPowerLevel a struct is simple but I can't make structs out of floats (Probably skill issue on my part). I assume it would be bad practice to put the different power levels / ranges inside the system itself (Specifically in GetCurrentRange and GetCurrentWattage).

My only other idea was to do something like this:

```
public static class RadioJamWattage
{
    public const float HighWatt = 12f;
    public const float MediumWatt = 2f;
    public const float LowWatt = 1f;
}
```

What do you think?

Reply:
> (Also I'm not sure this works either. I've tried but it doesn't seem to like it...)

Damn. In C# enums can't have float values. Despair

Reply:
Mannn that is the smart way of doing it! I'm not very good so it might take me a while but I'll implement it now :)

Thanks!

Reply:
Did it! Let me know if I did it correctly. For the for loop in the verbs I just used a standard one. However, I also did it with a for each and wasn't sure which is better!

```
byte index = 0;
foreach (var setting in entity.Comp.Settings)
{
    byte currIndex = index;
    var verb = new Verb
    {
        Priority = currIndex,
        Category = VerbCategory.PowerLevel,
        Disabled = entity.Comp.SelectedPowerLevel == currIndex,
        Act = () =>
        {
            entity.Comp.SelectedPowerLevel = currIndex;
            _popup.PopupEntity(Loc.GetString(setting.Message), user, user);
        },
        Text = Loc.GetString(setting.Name),
    };
    args.Verbs.Add(verb);
    index++;
}
```
I have only two issues with the for each:
1.) Storing the index separately seems weird
2.) I'm not sure if the iterator guarantees that it goes through the list in index order.

Reply:
> 

1) Yeah that does looks kinda weird, but it looks like that's how most people do it to keep it "simple". 
2) It should go through the list in index order.

Reply:
Done!